aggsender/flows/factory_test.go

```diff
@@ -79,7 +101,17 @@ func TestNewFlow(t *testing.T) {
 			mockL2BridgeSyncer := new(mocks.L2BridgeSyncer)
 			mockL2BridgeSyncer.EXPECT().OriginNetwork().Return(1)
 			mockLogger := log.WithFields("test", "NewFlow")
+			funcNewEVMChainGERReader = func(_ common.Address, _ aggoracletypes.EthClienter) (*chaingerreader.EVMChainGERReader, error) {
+				return &chaingerreader.EVMChainGERReader{}, nil
+			}
+			funcGetL2StartBlock = func(_ common.Address, _ types.EthClient) (uint64, error) {
+				return 100, nil
+			}
 
+			mockL1Client.EXPECT().CallContract(mock.Anything, mock.Anything, mock.Anything).Return([]byte{1, 2, 3}, nil).Maybe()
```

Comment:
Maybe you could do the same for `NewOptimisticModeQuerierFromContract` that you did for `var funcGetL2StartBlock = getL2StartBlock` and not have these mocks?